File: src/base/ApiClient.ts
import "dotenv/config";
import { ApiClientBase } from "./ApiClientBase.js";

export class ApiClient extends ApiClientBase {
  private static classInstance?: ApiClient;

  private constructor() {
    super();
  }

  public static getInstance() {
    if (!this.classInstance) {
      this.classInstance = new ApiClient();
    }

    return this.classInstance;
  }
}

File: src/base/ApiClient.py
from src.base.ApiClientBase import ApiClientBase

class ApiClient(ApiClientBase):
    _instance = None

    def __new__(cls, *args, **kwargs):
        if not cls._instance:
            cls._instance = super(ApiClient, cls).__new__(cls, *args, **kwargs)
        return cls._instance

    def __init__(self):
        super().__init__()
